runtime/gc_vlhgc/GlobalMarkingScheme.cpp

```diff
@@ -763,6 +764,49 @@ MM_GlobalMarkingScheme::scanPointerArrayObject(MM_EnvironmentVLHGC *env, J9Index
 	}
 }
 
+void
+MM_GlobalMarkingScheme::doStackSlot(MM_EnvironmentVLHGC *env, J9Object *fromObject, J9Object** slotPtr, J9StackWalkState *walkState, const void *stackLocation)
+{
+	J9Object *object = *slotPtr;
+	if (isHeapObject(object)) {
+		/* heap object - validate and mark */
+		Assert_MM_validStackSlot(MM_StackSlotValidator(0, *slotPtr, stackLocation, walkState).validate(env));
+		markObject(env, object);
+		rememberReferenceIfRequired(env, fromObject, object);
+	} else if (NULL != object) {
+		/* stack object - just validate */
+		Assert_MM_validStackSlot(MM_StackSlotValidator(MM_StackSlotValidator::NOT_ON_HEAP, *slotPtr, stackLocation, walkState).validate(env));
+	}
+}
+
+void
+stackSlotIteratorForGlobalMarkingScheme(J9JavaVM *javaVM, J9Object **slotPtr, void *localData, J9StackWalkState *walkState, const void *stackLocation)
+{
+	StackIteratorData4GlobalMarkingScheme *data = (StackIteratorData4GlobalMarkingScheme *)localData;
+	data->globalMarkingScheme->doStackSlot(data->env, data->fromObject, slotPtr, walkState, stackLocation);
+}
+
+void
+MM_GlobalMarkingScheme::scanContinuationObject(MM_EnvironmentVLHGC *env, J9Object *objectPtr, ScanReason reason)
+{
+	scanMixedObject(env, objectPtr, reason);
+	J9VMThread *currentThread = (J9VMThread *)env->getLanguageVMThread();
+	if (VM_VMHelpers::needScanStacksForContinuation(currentThread, objectPtr)) {
+		StackIteratorData4GlobalMarkingScheme localData;
+		localData.globalMarkingScheme = this;
+		localData.env = env;
+		localData.fromObject = objectPtr;
+		bool bStackFrameClassWalkNeeded = false;
+#if defined(J9VM_GC_DYNAMIC_CLASS_UNLOADING)
+		if (isDynamicClassUnloadingEnabled()) {
```

Comment:
Not sure myself, but looking at 

https://github.com/eclipse-openj9/openj9/blob/master/runtime/gc_vlhgc/GlobalMarkingScheme.cpp#L1100

this should probably be 

bStackFrameClassWalkNeeded = ! _classDataAsRoots;

Reply:
`			_classDataAsRoots = MM_GCExtensions$DynamicClassUnloading.DYNAMIC_CLASS_UNLOADING_NEVER == _extensions.dynamicClassUnloading();
` so _classDataAsRoots = false when isDynamicClassUnloadingEnabled()

Reply:
Right, at the end it's the same, but since _classDataAsRoots is  flag in roots scanner, my proposal is more complicated. You were correct basing it on  isDynamicClassUnloadingEnabled(), but as I mentioned before, unify the code (don't use 'if')